extensions/wikidata/module/scripts/dialogs/manage-account-dialog.js

```diff
@@ -2,120 +2,210 @@ var ManageAccountDialog = {};
-      },
-      "json"
-  );
-};
+ManageAccountDialog.display = function (logged_in_username, callback) {
 
-ManageAccountDialog.display = function(logged_in_username, saved_credentials, callback) {
-  var self = this;
-  var frame = $(DOM.loadHTML("wikidata", "scripts/dialogs/manage-account-dialog.html"));
-  var elmts = this._elmts = DOM.bind(frame);
-
-  this._elmts.dialogHeader.text($.i18n('wikidata-account/dialog-header'));
-  this._elmts.explainLogIn.html($.i18n('wikidata-account/explain-log-in'));
-  this._elmts.usernameLabel.text($.i18n('wikidata-account/username-label'));
-  this._elmts.usernameInput.attr("placeholder", $.i18n('wikidata-account/username-placeholder'));
-  this._elmts.passwordLabel.text($.i18n('wikidata-account/password-label'));
-  this._elmts.passwordInput.attr("placeholder", $.i18n('wikidata-account/password-placeholder'));
-  this._elmts.rememberCredentialsLabel.text($.i18n('wikidata-account/remember-credentials-label'));
-  this._elmts.dialogHeader.text($.i18n('wikidata-account/dialog-header'));
-  this._elmts.cancelButton1.text($.i18n('wikidata-account/close'));
-  this._elmts.cancelButton2.text($.i18n('wikidata-account/close'));
-  this._elmts.loggedInAs.text($.i18n('wikidata-account/logged-in-as'));
-  this._elmts.logoutButton.text($.i18n('wikidata-account/log-out'));
-  this._elmts.loginButton.val($.i18n('wikidata-account/log-in'));
+  if (logged_in_username == null) {
+    logged_in_username = ManageAccountDialog.tryLoginWithCookies(callback);
+  }
 
   if (logged_in_username != null) {
-      elmts.loginArea.remove();
+    ManageAccountDialog.displayLoggedIn(logged_in_username, callback);
   } else {
-      elmts.logoutArea.remove();
+    ManageAccountDialog.displayPasswordLogin(callback);
   }
+};
+
+ManageAccountDialog.tryLoginWithCookies = function (callback) {
+  var logged_user_name = null;
+  $.ajaxSetup({async: false});
+  Refine.postCSRF(
+      "command/wikidata/login",
+      {},
+      function (data) {
+        if (data.logged_in) {
+          callback(data.username);
+          logged_user_name = data.username;
+        } else {
+          logged_user_name = null;
+        }
+      });
+  $.ajaxSetup({async: true});
+  return logged_user_name;
+};
 
-  this._level = DialogSystem.showDialog(frame);
-  this._elmts.usernameInput.focus();
+ManageAccountDialog.initCommon = function (elmts) {
+  elmts.dialogHeader.text($.i18n('wikidata-account/dialog-header'));
+  elmts.explainLogIn.html($.i18n('wikidata-account/explain-log-in'));
+  elmts.cancelButton.text($.i18n('wikidata-account/close'));
+};
 
-  var dismiss = function() {
-    DialogSystem.dismissUntil(self._level - 1);
+ManageAccountDialog.displayLoggedIn = function (logged_in_username, callback) {
+  var frame = $(DOM.loadHTML("wikidata", "scripts/dialogs/logged-in-dialog.html"));
+  var elmts = DOM.bind(frame);
+  ManageAccountDialog.initCommon(elmts);
+  elmts.loggedInAs.text($.i18n('wikidata-account/logged-in-as'));
```

Comment:
This should include the username as a parameter if it's not too complex to manage with the rest of the HTML.